pkg/block/metadata/meta.go

```diff
@@ -59,6 +60,9 @@ type Thanos struct {
 
 	// Source is a real upload source of the block.
 	Source SourceType `json:"source"`
+
+	// LastModified records the last modified time of meta.json
+	LastModified time.Time `json:"lastModified,omitempty"`
```

Comment:
Should be serialized to JSON? I don't think so. This field should be populated only in-memory, so I would not serialize it in the JSON.